Jenkinsfile

```diff
@@ -154,13 +162,56 @@ node {
       if (params.BUILD_GRAVITY_APP) {
         withEnv(MAKE_ENV) {
           sh 'make export'
-          archiveArtifacts "build/application.tar"
         }
       } else {
         echo 'skipped build gravity app'
       }
     }
+
+    stage('upload application tarball to S3') {
+      if (isProtectedBranch(env.TAG) && params.PUBLISH_APP_PACKAGE) {
+        withCredentials([usernamePassword(credentialsId: "${AWS_CREDENTIALS}", usernameVariable: 'AWS_ACCESS_KEY_ID', passwordVariable: 'AWS_SECRET_ACCESS_KEY')]) {
+          def artifactName = "application.tar"
+          def s3AppName = "stolon-app"
+          if (params.BUILD_GRAVITY_HELM_APP) {
+            artifactName = "helm-application.tar"
+            s3AppName = "stolon-app-helm"
+          }
+          def s3Url = "s3://${S3_UPLOAD_PATH}/${s3AppName}:${APP_VERSION}.tar"
+          withEnv(MAKE_ENV + ["S3_URL=${s3Url}", "artifactName=${artifactName}"]) {
```

Comment:
just a detail, but is there a reason to send S3_URL and artifactName as env variables for sh to expand, you can use double quotes and replace in the command as you did for s3Url